File: 1116-maximum-level-sum-of-a-binary-tree/1116-maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        if root.left is None and root.right is None:
            return 1
        def BFS(root):
            q=deque([root])
            maximum=[]
            maximum.append(root.val)
            while q:
                l=len(q)
                value=0
                for i in range(l):
                    tempRoot=q.popleft()
                    if tempRoot.left:
                        q.append(tempRoot.left)
                        value+=tempRoot.left.val
                    if tempRoot.right:
                        q.append(tempRoot.right)
                        value+=tempRoot.right.val
                maximum.append(value)
            maximum.pop()
            print(maximum)
            #print(list(q))
            return maximum.index(max(maximum))+1
        return BFS(root)
        

File: 1116-maximum-level-sum-of-a-binary-tree/1116-maximum-level-sum-of-a-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int BFS(TreeNode root){
        int i,l;
        Queue<TreeNode> q=new LinkedList<>();
        List<Integer> maximum=new ArrayList<>();
        maximum.add(root.val);
        q.add(root);
        while(!q.isEmpty()){
            l=q.size();
            int value=0;
            for(i=0;i<l;i++){
                TreeNode r=q.poll();
                if(r.left!=null){
                    value+=r.left.val;
                    q.add(r.left);
                }
                if(r.right!=null){
                    value+=r.right.val;
                    q.add(r.right);
                }
            }
            maximum.add(value);
        }
        maximum.remove(maximum.size()-1);
        int maxNum = Collections.max(maximum);
        return maximum.indexOf(maxNum) + 1;
    }
    public int maxLevelSum(TreeNode root) {
        if(root.left==null && root.right==null){
            return 1;
        }
        return BFS(root);
    }
}